src/main/scala/abalone/Pos.scala

```diff
@@ -190,8 +190,38 @@ case class Pos private (index: Int) extends AnyVal {
   def right: Option[Pos]     = Pos.at(file.index + 1, rank.index)
   def downRight: Option[Pos] = Pos.at(file.index, rank.index - 1)
   def upRight: Option[Pos]   = Pos.at(file.index + 1, rank.index + 1)
+
   def neighbours: List[Option[Pos]] = List(left, downLeft, upLeft, right, downRight, upRight)
 
+  // NOTE - *neighbourhood
+  // these below only work for neighbour pos but that's probably fine as in Abalone we only move to (potentially extended) neighbourhood
+  def dir(dir: Option[String]): Option[Pos]  = {
+    dir match {
+      case Some("left") => this.left
+      case Some("downLeft") => this.downLeft
+      case Some("upLeft") => this.upLeft
+      case Some("right") => this.right
+      case Some("downRight") => this.downRight
+      case Some("upRight") => this.upRight
+      case _ => None
+    }
+  }
+  def dir(pos: Pos): Option[String] = {
+    (pos.file.index - this.file.index, pos.rank.index - this.rank.index) match {
+      case (0, 1) => Some("upLeft")
```

Comment:
can you double check the coordinates it doesn't seem right to me, right and left swapping whether they are negative or not.

Reply:
disregard this, it looks correct upon me looking at the diagram, just felt wrong but that the new coordinate system for you.

Reply:
I do not understand I think
I tried this in console :
```
scala> strategygames.abalone.Pos.A1.dir(strategygames.abalone.Pos.B1)
val res2: Option[String] = Some(right)

scala> strategygames.abalone.Pos.B1.dir(strategygames.abalone.Pos.A1)
val res3: Option[String] = Some(left)
```

Reply:
ah ok :+1: